internal/lldp/rx.go

```diff
@@ -213,23 +306,61 @@ func (l *LLDP) handlePacketCDP(mac string, raw *layers.CiscoDiscovery, info *lay
 	}
 
 	if len(info.MgmtAddresses) > 0 {
-		n.ManagementAddress = string(info.MgmtAddresses[0])
+		ip := info.MgmtAddresses[0]
+		ipFamily := "ipv4"
+		if ip.To4() == nil {
+			ipFamily = "ipv6"
+		}
+
+		l.l.Info().
+			Str("ip", ip.String()).
+			Str("ip_family", ipFamily).
+			Interface("ip", ip).
+			Interface("info", info).
+			Msg("parsed IP address")
+
+		n.ManagementAddress = &ManagementAddress{
+			AddressFamily:    ipFamily,
+			Address:          ip.String(),
+			InterfaceSubtype: "if_name",
+			InterfaceNumber:  0,
+			OID:              "",
+		}
 	}
 
-	l.addNeighbor(mac, *n, ttl)
+	l.addNeighbor(n, ttl)
 
 	return nil
 }
 
-func (l *LLDP) shutdownCapture() error {
-	if l.tPacket != nil {
-		l.tPacket.Close()
-		l.tPacket = nil
+func (l *LLDP) stopCapture() error {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+
+	if !l.rxRunning {
+		return nil // Already stopped
+	}
+
+	logger := l.l.With().Str("interface", l.interfaceName).Logger()
+	logger.Info().Msg("stopping LLDP receiver")
+
+	// Cancel context to signal stop
+	rxCancel := l.rxCancel
+	if rxCancel != nil {
+		rxCancel()
+		l.rxCancel = nil
 	}
 
-	if l.pktSource != nil {
-		l.pktSource = nil
+	if l.tPacketRx != nil {
+		l.tPacketRx.Close()
+		l.tPacketRx = nil
 	}
 
+	if l.pktSourceRx != nil {
+		l.pktSourceRx = nil
+	}
+
+	time.Sleep(100 * time.Millisecond)
```

Comment:
... or if not, perhaps this sleep should be moved up to before we start closing and `nil`ing things out so in-flight stuff has a chance to notice and cancel the loop?